src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -868,3 +867,43 @@ def perturb_household_age(self, simulants: pd.DataFrame) -> pd.Series:
         simulants.loc[simulants["age"] > 99, "age"] = 99
 
         return simulants["age"]
+
+    def perturb_individual_age(self, pop: pd.DataFrame) -> pd.Series:
+        # Takes dataframe containing a column "age" and returns a series of ages shifted with truncnorm mean=0 and
+        #  sd=10 years.  If a simulant's age shift results in a negative age their perturbed age will be redrawn from
+        #  the distribution.
+        # pop will be the population for group quarters or immigrants migrating to the US.
+        # todo: Consider changing this function to build a distribution for each simulan instead of using recursion.
+
+        def perturb_age(ages_to_shift: pd.Series, additional_key: str) -> pd.Series:
+            age_shift_propensity = self.randomness.get_draw(
+                ages_to_shift.index,
+                additional_key=additional_key,
+            )
+            # Convert to truncnorm distribution with mean=0 and sd=10
+            age_shift = pd.Series(
+                data=stats.norm.ppf(age_shift_propensity, loc=0, scale=10),
+                index=age_shift_propensity.index,
+            )
+            return age_shift
+
+        # Get age shift and redraw for simulants who get negative ages
+        to_shift = pd.Series(True, index=pop.index)
+        while_key = 1
+        while to_shift.any():
```

Comment:
I'd suggest a `for` loop with some max amount of retries to provide a guarantee against an infinite loop. You will also get your iteration count for free.